Project/src/MakeCall/FunctionalStreamRenderer.js

```diff
@@ -80,29 +80,29 @@ export const FunctionalStreamRenderer = forwardRef(({ remoteParticipant, stream,
     }
     const isReceivingChanged = () => {
         try {
-            if (stream.isAvailable) {
-                const isReceiving = stream.isReceiving;
-                if (!isReceiving && !isLoading) {
-                    setIsLoading(true)
-                } else if (isReceiving && isLoading) {
-                    setIsLoading(false);
-                }
-            }
+            setIsLoading(stream.isAvailable && !stream.isReceiving);
         } catch (e) {
             console.error(e);
         }
     };
 
+    /*
+     * Returns the amount of videos currently rendered excluding the ScreenSharing
+     */
+    const getCurrentNumberOfRenderers = () => {
+        return document.querySelector('.video-grid-row').querySelectorAll('.stream-container:not([id*="ScreenSharing"]) video').length;
+    };
+
     const isAvailableChanged = async () => {
         try {
             if (dominantSpeakerMode && dominantRemoteParticipant !== remoteParticipant) {
                 return;
             }
-            if (call.activeRemoteVideoStreamViews?.size >= maximumNumberOfRenderers && !stream.isAvailable) {
+            if (getCurrentNumberOfRenderers() >= maximumNumberOfRenderers && !stream.isAvailable) {
                 updateStreamList();
             }
             if (stream.isAvailable && !renderer) {
-                if (call.activeRemoteVideoStreamViews?.size >= maximumNumberOfRenderers) {
+                if (getCurrentNumberOfRenderers() >= maximumNumberOfRenderers) {
```

Comment:
cache the value from previous call to avoid paying that cost twice

Reply:
not applicable anymore